Objects/typeobject.c

```diff
@@ -8801,8 +8801,15 @@ wrap_ternaryfunc_r(PyObject *self, PyObject *args, void *wrapped)
 
     /* Note: This wrapper only works for __rpow__() */
 
-    if (!PyArg_UnpackTuple(args, "__rpow__", 1, 2, &other, &third))
+    Py_ssize_t size = check_pow_args(args);
+    if (size == -1) {
         return NULL;
+    }
+    other = PyTuple_GET_ITEM(args, 0);
+    if (size == 2) {
+       third = PyTuple_GET_ITEM(args, 1);
+    }
```

Comment:
This could also be moved into `check_pow_args` (or rather `unpack_pow_args`).

BTW, it is still a question whether `__rpow__` and `__ipow__` should support the third argument. Normally they never called with three arguments. So it may be better to leave the current PR code, it will be easier to change in future.

Reply:
Yes, we can open a new issue about that. For now, we only change the error message, not any logical parts. Thanks a lot for the review! 👍